docker-compose.yml

```diff
@@ -0,0 +1,46 @@
+version: '3.8'
+
+services:
+  traefik:
+    image: traefik:v3.0
+    command:
+      - "--api.insecure=true"
+      - "--providers.docker=true"
+      - "--providers.docker.exposedbydefault=false"
+      - "--entrypoints.web.address=:8000"
+    ports:
+      - "8000:8000"
+      - "8090:8080"  # Traefik dashboard
+    volumes:
+      - "/var/run/docker.sock:/var/run/docker.sock:ro"
+    privileged: true
+
+  redis:
+    image: redis:7-alpine
+    ports:
+      - "6379:6379"
+    volumes:
+      - ./redis/redis.conf:/usr/local/etc/redis/redis.conf
+    command: redis-server /usr/local/etc/redis/redis.conf
+    healthcheck:
+      test: ["CMD", "redis-cli", "ping"]
+      interval: 5s
+      timeout: 3s
+      retries: 5
+
+  streaming_api:
+    build: .
+    environment:
+      - REDIS_URL=redis://redis:6379/0
+    depends_on:
+      redis:
+        condition: service_healthy
+    volumes:
+      - ./server.py:/app/server.py:ro
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.streaming_api.rule=Host(`localhost`)"
+      - "traefik.http.routers.streaming_api.entrypoints=web"
+      - "traefik.http.services.streaming_api.loadbalancer.server.port=8000"
+    deploy:
+      replicas: 3
```

Comment:
I've seen this, but I'm actually not convinced it works. I haven't found a definitive answer on that question though.

Did you test get unique server headers from different responses?

Reply:
I was getting this output:
```
for i in {1..20}; do curl -s -D - http://localhost:8000/stream/live 2>/dev/null | grep X-Server-Host; done | sort
X-Server-Host: 002586e99df0
X-Server-Host: 002586e99df0
X-Server-Host: 002586e99df0
X-Server-Host: 002586e99df0
X-Server-Host: 002586e99df0
X-Server-Host: 002586e99df0
X-Server-Host: 002586e99df0
X-Server-Host: 07cfda44c664
X-Server-Host: 07cfda44c664
X-Server-Host: 07cfda44c664
X-Server-Host: 07cfda44c664
X-Server-Host: 07cfda44c664
X-Server-Host: 07cfda44c664
X-Server-Host: 07cfda44c664
X-Server-Host: 3dd252d16556
X-Server-Host: 3dd252d16556
X-Server-Host: 3dd252d16556
X-Server-Host: 3dd252d16556
X-Server-Host: 3dd252d16556
X-Server-Host: 3dd252d16556
```